src/Util/PrestaShopUtils.php

```diff
@@ -71,16 +122,30 @@ public function getLocalVersions(): array
             if (in_array($prestaShopVersion, $exclude) || !is_dir($this->prestaShopDir . '/' . $prestaShopVersion)) {
                 continue;
             }
-            $versions[] = new PrestaShop($prestaShopVersion);
+            $prestashop = new PrestaShop($prestaShopVersion);
+            $this->setVersionsCompat($prestashop);
+            $prestashop->setDownloadUrl($this->publicDownloadUrlProvider->getPrestaShopDownloadUrl($prestaShopVersion));
+            $versions[] = $prestashop;
         }
 
         return $versions;
     }
 
-    public function setVersionsCompat(PrestaShop $prestaShop): void
+    private function setVersionsCompat(PrestaShop $prestaShop): void
     {
         $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
-        if (!$content = file_get_contents($this->prestaShopDir . '/' . $prestaShop->getVersion() . '/install_version.php')) {
+        $installZip = new ZipArchive();
+        $installZip->open($this->prestaShopDir . '/' . $prestaShop->getVersion() . '/prestashop.zip');
+        $installZip->extractTo($this->prestaShopDir . '/' . $prestaShop->getVersion() . '/prestashop/');
```

Comment:
I think we might need try/catch to handle when the archive process fails

Reply:
I'm not sure we want the script to continue if this fails actually 🤔

Reply:
Even if we want the script to stop on such failure, then I would prefer we make it explicit:
```
try {
} catch (CannotCreateZipArchiveException $exception) {
    throw new Exception('App cannot Create a ZipArchive, please debug');
}
```